cmd/pack/main.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+
+	"github.com/spf13/pflag"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/crossplane/crossplane-cli/pkg/pack"
+	"github.com/crossplane/crossplane/apis/workload/v1alpha1"
+)
+
+type packOptions struct {
+	help      bool
+	useFile   string
+	name      string
+	namespace string
+}
+
+var pOpts = &packOptions{}
+
+func init() {
+	pflag.BoolVarP(&pOpts.help, "help", "h", false, "Shows this help message")
+	pflag.StringVarP(&pOpts.useFile, "file", "f", "", "Specifies file path to read from")
+	pflag.StringVar(&pOpts.name, "name", fmt.Sprintf("app-%s", fmt.Sprint(rand.Intn(50000))), "Name")
+	pflag.StringVarP(&pOpts.namespace, "namespace", "n", "default", "Namespace")
+	pflag.Parse()
+}
+
+func main() {
+	data, err := runPack(pOpts)
+	if err != nil {
+		panic(err)
+	}
+
+	if data != nil {
+		// TODO(hasheddan): allow writing output to path using flag
+		fmt.Print(string(data))
+	}
+
+	os.Exit(0)
+}
+
+func runPack(opts *packOptions) ([]byte, error) {
+	if opts.help {
```

Comment:
Why not move this `help` related logic up to `main`?

Reply:
I generally try to keep everything out of main for unit testing purposes :)

Reply:
Hmm, I'm not sure I can see how it helps wit unit tests. Personally, it doesn't make much sense to me for `runPack` function's output to be printing help and return. The pattern I see in our `main.go` files is that all CLI flags, env vars and CLI-specific stuff like help are handled in the `main` function and the rest of the functions take parameters.

Reply:
It is difficult to test any logic that is in `main` but easy to test it if `main` passes args to a different function. I think the areas where we do not do this in Crossplane should be updated honestly. You can read more about this here: https://pace.dev/blog/2020/02/12/why-you-shouldnt-use-func-main-in-golang-by-mat-ryer

Reply:
Got it. Instead of abstraction, what I find useful is putting the logic that doesn't need test in the main function and call actual logic functions that exist in `pkg` folder. For example, if it's a bunch of [CLI-specific operation](https://dave.cheney.net/practical-go/presentations/qcon-china.html#_keep_package_main_small_as_small_as_possible) or help, I'd put them in `main` because when I write tests for those direct and simple calls, I find myself testing the CLI library instead of my own logic. Instead I opt for testing them in integration test to make sure flags and env vars are still consistent as interface of my program.

Both approaches are just fine though.